app/src/ui/stashing/stash-diff-viewer.tsx

```diff
@@ -109,14 +109,11 @@ const Header: React.SFC<{
   dispatcher: Dispatcher
   isWorkingTreeClean: boolean
 }> = props => {
-  const { dispatcher, repository, stashEntry, isWorkingTreeClean } = props
-
-  const onClearClick = () => {
-    dispatcher.dropStash(repository, stashEntry)
+  const onDiscardClick = () => {
+    props.dispatcher.dropStash(props.repository, props.stashEntry)
```

Comment:
`dispatcher`, `repository`, and `stashEntry` can be destructured to reduce the amount of `props.<field>` expressions.

Reply:
I addressed this in c8956f0 since I broke the build with my previous merge.